e2e/portalicious/pages/TableComponent.ts

```diff
@@ -0,0 +1,64 @@
+import { Locator } from '@playwright/test';
+import { Page } from 'playwright';
+
+import BasePage from './BasePage';
+
+class TableComponent extends BasePage {
+  readonly page: Page;
+  readonly tableRows: Locator;
+  readonly goToProfileOption: Locator;
+
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+    this.tableRows = this.page.locator('table tbody tr');
+    this.goToProfileOption = this.page.getByText('Go to profile');
+  }
+
+  async getCell(row: number, column: number) {
+    return this.tableRows.nth(row).locator('td').nth(column);
+  }
+
+  async getFirstRegistrationNameFromTable() {
```

Comment:
Similar to what I said above. This function should probably be defined in `RegistrationsPage`, but this it would leverage the table with a usage along these lines:

```ts
async getFirstRegistrationNameFromTable() {
Comment


Similar to what I said above. This function should probably be defined in `RegistrationsPage`, but this it would leverage the table with a usage along these lines:

```ts
async getFirstRegistrationNameFromTable() {
    await this.table.waitForLoad(); // this function would do the waitForTimeout and waitForSelector
    const fullName = await this.table.getCell(0, 2);
    const fullNameText = (await fullName.textContent())?.trim();
    return fullNameText;
  }